src/extension/executors/runner/index.ts

```diff
@@ -514,94 +516,18 @@ export async function resolveRunProgramExecution(
   // todo(sebastian): removing $-prompts from shell scripts should move kernel-side
   const rawCommands = prepareCommandSeq(script, languageId)
   const result = await resolver.resolveProgram(rawCommands, runnerEnv?.getSessionId())
-
-  const exportMatches = result.response?.vars.map((v) => {
-    return <CommandExportExtractMatch>{
-      type: v.status !== ResolveProgramResponse_Status.RESOLVED ? 'prompt' : 'direct',
-      key: v.name,
-      value: v.resolvedValue || v.originalValue || 'Enter a value please',
-      match: v.name,
-      hasStringValue: v.status === ResolveProgramResponse_Status.UNRESOLVED_WITH_PLACEHOLDER,
-      isPassword: v.status === ResolveProgramResponse_Status.UNRESOLVED_WITH_SECRET,
-    }
-  })
+  const vars = result.response.vars
 
   // todo(sebastian): once normalization is all kernel-side, it should return commands
   script = result.response.script ?? script
 
   // const commands = await parseCommandSeq(script, languageId, exportMatches, skipEnvs)
-  type CommandBlock =
-    | {
-        type: 'block'
-        content: string
-      }
-    | {
-        type: 'single'
-        content: string
-      }
-
-  const parsedCommandBlocks: CommandBlock[] = []
-
-  // todo(sebastian): do we still need this? should be handled server-side
-  // const skipEnvs = new Set([...(runnerEnv?.initialEnvs() ?? []), ...Object.keys(envs)])
-
-  let offset = 0
-
-  // todo(sebastian): align this with grpc types
-  for (const {
-    hasStringValue,
-    key,
-    match,
-    type,
-    value,
-    isPassword,
-    regexpMatch,
-  } of exportMatches) {
-    let userValue: string | undefined
-
-    let skip = false
-
-    switch (type) {
-      case 'prompt':
-        {
-          const userInput = await promptUserForVariable(
-            key,
-            value,
-            hasStringValue,
-            isPassword ?? false,
-          )
-
-          if (userInput === undefined) {
-            throw new Error('Cannot run cell due to canceled prompt')
-          }
-
-          userValue = userInput
-        }
-        break
-
-      case 'direct':
-        {
-          userValue = value
-        }
-        break
-
-      default: {
-        continue
-      }
-    }
-
-    if (regexpMatch) {
-      const prior = script.slice(offset, regexpMatch.index)
-      parsedCommandBlocks.push({ type: 'block', content: prior })
-      offset = regexpMatch.index + match.length
-    }
-
-    if (!skip && userValue !== undefined) {
-      parsedCommandBlocks.push({ type: 'single', content: `export ${key}="${userValue}"` })
-    }
+  const reducer = async (acc: Promise<CommandBlock[]>, current: VarResult) => {
```

Comment:
little nit-picky, but I think something more elaborate than `reducer` would be good here. E.g. `varToPromptReduer` or even `promptReducer`.